tests/workflow_task_reported_problems_test.go

```diff
@@ -0,0 +1,360 @@
+		searchValBytes := description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems]
+		searchVal, err := searchattribute.DecodeValue(searchValBytes, enumspb.INDEXED_VALUE_TYPE_KEYWORD_LIST, false)
+		require.NoError(t, err)
+		require.NotEmpty(t, searchVal)
+		require.Equal(t, "category=WorkflowTaskFailed", searchVal.([]string)[0])
+		require.Equal(t, "cause=WorkflowWorkerUnhandledFailure", searchVal.([]string)[1])
+
+		// Validate attempt number after verifying search attribute values
+		require.GreaterOrEqual(t, description.GetPendingWorkflowTask().Attempt, int32(2))
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Check if the search attributes are queryable
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		queriedWorkflows, err := s.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			Query:     "TemporalReportedProblems IS NOT NULL",
+			PageSize:  100,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(queriedWorkflows.Executions))
+
+		queriedWorkflows, err = s.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			Query:     "TemporalReportedProblems IN ('category=WorkflowTaskFailed', 'cause=WorkflowWorkerUnhandledFailure')",
+			PageSize:  100,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(queriedWorkflows.Executions))
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Unblock the workflow
+	s.shouldFail.Store(false)
+
+	var out string
+	err = workflowRun.Get(ctx, &out)
+
+	s.NoError(err)
+
+	// Validate the workflow completed successfully and the search attribute is removed
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.Equal(t, enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED, description.WorkflowExecutionInfo.Status)
+		require.Nil(t, description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems])
+	}, 5*time.Second, 500*time.Millisecond)
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) TestWFTFailureReportedProblems_SetAndClear_FailAfterActivity() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	s.OverrideDynamicConfig(dynamicconfig.NumConsecutiveWorkflowTaskProblemsToTriggerSearchAttribute, 2)
+	s.shouldFail.Store(true)
+
+	s.Worker().RegisterWorkflow(s.simpleWorkflow)
+	s.Worker().RegisterActivity(s.simpleActivity)
+
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        testcore.RandomizeStr("wf_id-" + s.T().Name()),
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, s.simpleWorkflow)
+	s.NoError(err)
+
+	// Make sure the workflow has started and had an activity task scheduled and finished
```

Comment:
I'm going to repeat the comment from above here, but I won't do that for the rest of the comments that need to be pasted for this test case, please apply the rest of the comments above.